avocado/loader.py

```diff
@@ -99,18 +100,34 @@ def _make_test(self, test_name, test_path, params, queue):
                     test_class = test.NotATest
                     test_parameters = test_parameters_queue
 
-        except ImportError, details:
+        # Since a lot of things can happen here, the broad exception is
+        # justified. The user will get it unadulterated anyway, and avocado
+        # will not crash.
+        except Exception, details:
             if os.access(test_path, os.X_OK):
                 # Module can't be imported, and it's executable. Let's try to
                 # execute it.
                 test_class = test.SimpleTest
                 test_parameters = test_parameters_simple
             else:
-                # Module can't be imported and it's not an executable. Our
-                # best guess is that this is a buggy test.
-                test_class = test.BuggyTest
-                params['exception'] = details
+                # Module can't be imported and it's not an executable. Let's
+                # see if there's an avocado import into the test. Although
+                # not entirely reliable, we hope it'll be good enough.
+                likely_avocado_test = False
+                with open(test_path, 'r') as test_file_obj:
+                    test_contents = test_file_obj.read()
+                    patterns = ['from avocado.* import', 'import avocado.*']
```

Comment:
I would better use `^from...` and `^import....`, to restrict the case where this line is present inside a string, and this is not a avocado test. Most of the time, for a test, we use imports in the column 0.